From pavel@pavlinux.ru  Thu Jan 14 17:18:52 2010
From: Pavel Vasilyev <pavel@pavlinux.ru>
Date: Mon, 11 Jan 2010 21:08:02 +0300
Subject: Staging: dream: HTC Dream camera, not need sizeof
To: linux-kernel@vger.kernel.org, drivers_staging@kernel-bugs.osdl.org, greg@kroah.com
Message-ID: <201001112108.02634.pavel@pavlinux.ru>


http://bugzilla.kernel.org/show_bug.cgi?id=14825

sizeof(extlen), always will be sizeof( unit32_t) or 4
It seems that something is wrong?!?!

Signed-off-by: Pavel Vasilyev <pavel@pavlinux.ru>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/dream/camera/msm_vfe7x.c |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

--- a/drivers/staging/dream/camera/msm_vfe7x.c
+++ b/drivers/staging/dream/camera/msm_vfe7x.c
@@ -255,8 +255,7 @@ static int vfe_7x_init(struct msm_vfe_ca
 
 	extlen = sizeof(struct vfe_frame_extra);
 
-	extdata =
-		kmalloc(sizeof(extlen), GFP_ATOMIC);
+	extdata = kmalloc(extlen, GFP_ATOMIC);
 	if (!extdata) {
 		rc = -ENOMEM;
 		goto init_fail;
